bundles/CoreBundle/src/EventListener/Frontend/ElementListener.php

```diff
@@ -183,10 +184,16 @@ private function handleAdminUserDocumentParams(Request $request, ?Document $docu
         return $document;
     }
 
-    protected function handleEditmode(Document $document, User $user, SessionInterface $session): Document
+    protected function handleEditmode(
+        Document $document,
+        User $user,
+        SessionInterface $session,
+        bool $isPimcoreStudio
+    ): Document
     {
-        // check if there is the document in the session
-        if ($documentFromSession = Document\Service::getElementFromSession('document', $document->getId(), $session->getId())) {
+        // check if there is the document in the session (for admin classic UI)
+        $documentFromSession = Document\Service::getElementFromSession('document', $document->getId(), $session->getId());
+        if (!$isPimcoreStudio && $documentFromSession) {
```

Comment:
just to be sure, when it's pimcoreStudio it doesn't check anymore if there's a document in the session but it still has to get the latestVersion (as per the `else` block) ? 

In other words, doesn't it already use the latest document already? Feels like something might have changed and should be like

```suggestion
        if ($isPimcoreStudio){ 
           return $document;
        }
        $documentFromSession = Document\Service::getElementFromSession('document', $document->getId(), $session->getId());
```

Reply:
if it still has to get the latest version, i'd rewrite as
```
        if (!$isPimcoreStudio){
              // check if there is the document in the session (for admin classic UI)
              $documentFromSession = Document\Service::getElementFromSession('document', $document->getId(), $session->getId());
              if ($documentFromSession) {
                  // if there is a document in the session use it
                  $this->logger->debug('Loading editmode document {document} from session', [
                      'document' => $document->getFullPath(),
                  ]);
+                 return $documentFromSession;
              } 

       }

//the former ELSE part

      $this->logger->debug('Loading editmode document {document} from latest version', [
          'document' => $document->getFullPath(),
      ]);

      // set the latest available version for editmode if there is no doc in the session
      if ($document instanceof Document\PageSnippet) {
          $latestVersion = $document->getLatestVersion($user->getId());
          if ($latestVersion) {
              $latestDoc = $latestVersion->loadData();

              if ($latestDoc instanceof Document\PageSnippet) {
                  $document = $latestDoc;
              }
          }
      }


     return $document;

```
which avoids an unnecessary TmpStore get (db call,unserialize) & check expire in `Service::getElementFromSession()` from Studio pov

Reply:
We still need to get the latest version - we only can skip the document from the session. Yes your change would slightly optimize if if working in parallel with studio and classic UI.